mesheryctl/internal/cli/root/system/channel.go

```diff
@@ -45,90 +62,55 @@ var channelCmd = &cobra.Command{
 			channelName = channelSet
 		}
 
-		switch channelName {
+		channelNameSeperated := strings.Split(channelName, ":")
+
+		ContextContent, ok := mctlCfg.Contexts[focusedContext]
+		if !ok {
+			return errors.New("error while trying to fetch context content")
+		}
+
+		switch channelNameSeperated[0] {
 		case "stable":
-			if err := searchAndReplace(utils.DockerComposeFile, "edge", "stable"); err != nil {
-				log.Fatal("Error subscribing to release channel:", err)
-				return
+			ContextContent.Channel = channelNameSeperated[0]
+			if len(channelNameSeperated) == 1 {
+				ContextContent.Version = "latest"
+			} else {
+				ContextContent.Version = channelNameSeperated[1]
 			}
-			log.Info("Subscribed to stable channel.")
 
 		case "edge":
-			if err := searchAndReplace(utils.DockerComposeFile, "stable", "edge"); err != nil {
-				log.Fatal("Error subscribing to release channel:", err)
-				return
-			}
-
-			log.Info("Subscribed to edge channel.")
+			ContextContent.Channel = channelNameSeperated[0]
+			ContextContent.Version = "latest"
 
 		default:
-			currentChannel := getCurrentContextChannel()
+			currentChannel := mctlCfg.Contexts[focusedContext].Channel
 			if currentChannel == "" {
-				log.Fatal("No release channel subscription found. Please subscribe to either the 'stable' or 'edge' release channel.")
+				return errors.New("No release channel subscription found." +
+					"Please subscribe to either the 'stable' or 'edge' release channel")
 			}
-			log.Println("Channel subscription:", currentChannel)
 		}
 
-		if channelSwitch != "" {
-			if utils.IsMesheryRunning() {
-				if err := stop(); err != nil {
-					log.Fatal("Failed to stop Meshery:", err)
-					return
-				}
-			}
+		mctlCfg.Contexts[focusedContext] = ContextContent
+		viper.Set("contexts", mctlCfg.Contexts)
+		err = viper.WriteConfig()
+		if err != nil {
+			return err
+		}
+		PrintChannelAndVersionToStdout(mctlCfg.Contexts[focusedContext])
 
-			if err := start(); err != nil {
-				log.Fatal("Failed to start Meshery:", err)
-				return
+		if channelSwitch != "" {
+			err = startCmd.RunE(cmd, nil)
+			if err != nil {
+				return err
 			}
-			log.Info("Successfully switched channel...")
 		}
 
+		return nil
 	},
 }
 
 func init() {
 	channelCmd.Flags().StringVarP(&channelSet, "set", "s", "", "Release channel to be set for Meshery and its adapters.")
```

Comment:
The --version flag isn't supported here.

Reply:
I will add it

Reply:
Ohh Yea in this case I handled the case on if user specifies either stable or stable:version 
If you need a version flag seperately, I can get it done